web/html/src/manager/minion/ansible/schedule-playbook.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import { Messages, MessageType, Utils as MsgUtils } from "components/messages";
+import Network, { JsonResult } from "utils/network";
+import { Combobox, ComboboxItem } from "components/combobox";
+import { ActionChain, ActionSchedule } from "components/action-schedule";
+import { AsyncButton, Button } from "components/buttons";
+import { AnsiblePath } from "./ansible-path-type";
+import { InnerPanel } from "components/panels/InnerPanel";
+import { AceEditor } from "components/ace-editor";
+import { PlaybookDetails } from "./accordion-path-content";
+import { Utils } from "utils/functions";
+import { ActionChainLink, ActionLink } from "components/links";
+
+interface SchedulePlaybookProps {
+  playbook: PlaybookDetails,
+  onBack: () => void
+}
+
+export default function SchedulePlaybook({ playbook, onBack }: SchedulePlaybookProps) {
+  const [loading, setLoading] = useState(true);
+  const [playbookContent, setPlaybookContent] = useState("");
+  const [messages, setMessages] = useState<MessageType[]>([]);
+  const [inventoryPath, setInventoryPath] = useState<ComboboxItem | null>(null);
+  const [inventories, setInventories] = useState<string[]>([]);
+  const [actionChain, setActionChain] = useState<ActionChain | null>(null);
+  const [datetime, setDatetime] = useState(new Date());
+
+  // TODO: Use UserLocalization hook instead
+  const timezone = window.timezone;
+  const localTime = window.localTime;
+
+  useEffect(() => {
+    const getInventoryPaths = async () => {
```

Comment:
Small nitpick, here and elsewhere, if the whole async flow is handled by chaining `then`, `catch` etc then making the function `async` doesn't offer any additional benefit.

Reply:
oh right, the chain already returns a `Promise`. I'll remove `async` :+1: